pkg/apis/networking/types.go

```diff
@@ -362,3 +405,39 @@ type IngressBackend struct {
 	// Specifies the port of the referenced service.
 	ServicePort intstr.IntOrString
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClass represents the class of the Ingress, referenced by the Ingress
+// Spec.
+type IngressClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Controller is responsible for handling this class. This should be
+	// specified as a domain-prefixed path, e.g. "acme.io/ingress-controller".
+	// This allows for different "flavors" that are controlled by the same
+	// controller. For example, you may have different Parameters for the same
+	// implementing controller.
+	Controller string
+
+	// Parameters is a link to a custom resource configuration for the
+	// controller. This is optional if the controller does not require extra
+	// parameters.
+	// +optional
+	Parameters *api.ObjectReference
```

Comment:
[The KEP example](https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190125-ingress-api-group.md#ingress-class-proposal) uses a `*TypeLocalObjectReference` , but we use the `api.ObjectReference` here. I tried a quick `ack TypeLocalObjectReference` in the root of k/k and didn't find anything. Do you know if  `TypeLocalObjectReference` is supplanted by `api.ObjectReference` ?

Reply:
Not sure why I used a different type here. It looks like `api.ObjectReference` is the oldest of the reference types. One of the biggest differences is that with a local reference the parameters would have to live in the same namespace - that seems reasonable though.

Reply:
Update, after some clarification in relevant Service APIs discussion, TyledLocalObjectReference is preferred. I've updated the types to reflect that.